### PR TITLE
fix: Upgrade db encryption key handling

### PR DESCRIPTION
**Pull request by atomrc**
See https://github.com/wireapp/wire-desktop/pull/6423
and https://github.com/wireapp/wire-web-packages/pull/4885

This will basically fixe 2 things:
- ability to handle multiple system crypto (the desktop has a history of 3 different encryption system)
- fix issues with clear the DB when a permanent device is deleted from another device

### CODE REVIEW

File: src/script/storage/StorageRepository.ts [hunk 1]
```diff
@@ -32,18 +32,6 @@ export class StorageRepository {
...
-    return {
-      CRYPTOGRAPHY_TABLES: [
-        StorageSchemata.OBJECT_STORE.AMPLIFY,
-        StorageSchemata.OBJECT_STORE.CLIENTS,
-        StorageSchemata.OBJECT_STORE.KEYS,
-        StorageSchemata.OBJECT_STORE.SESSIONS,
-        StorageSchemata.OBJECT_STORE.PRE_KEYS,
-      ],
-    };
-  }
```
Comment (atomrc) [inline src/script/storage/StorageRepository.ts:45]: This is now handled by the core (in the logout method)

File: src/script/storage/StorageRepository.ts [hunk 2]
```diff
@@ -23,61 +23,23 @@ import {Logger, getLogger} from 'Util/Logger';
...
+    // Only used for testing purposes
+    await this.storageService.clearStores();
   }
 
   deleteDatabase(): Promise<boolean> {
     this.logger.warn(`Deleting database '${this.storageService.dbName}'`);
     return this.storageService.deleteDatabase();
   }
 
-  getValue(primaryKey: string): Promise<string | AmplifyRecord> {
```
Comment (atomrc) [inline src/script/storage/StorageRepository.ts:67]: was never consumed

File: src/script/storage/StorageRepository.ts [hunk 3]
```diff
@@ -23,61 +23,23 @@ import {Logger, getLogger} from 'Util/Logger';
...
-  }
-
-  clearStores(): Promise<void> {
-    return this.storageService
-      .clearStores()
-      .then(() => this.logger.info(`Cleared database '${this.storageService.dbName}'`));
   }
 
-  deleteCryptographyStores(): Promise<void> {
-    return this.storageService.deleteStores(StorageRepository.CONFIG.CRYPTOGRAPHY_TABLES);
```
Comment (atomrc) [inline src/script/storage/StorageRepository.ts:59]: was already implement in the core

File: src/script/storage/StorageRepository.ts [hunk 4]
```diff
@@ -23,61 +23,23 @@ import {Logger, getLogger} from 'Util/Logger';
...
-        return record.value;
-      }
-      throw new StorageError(StorageError.TYPE.NOT_FOUND, StorageError.MESSAGE.NOT_FOUND);
-    });
-  }
-
-  async saveValue<T>(primaryKey: string, value: T): Promise<string> {
-    await this.storageService.save(this.AMPLIFY_STORE_NAME, primaryKey, {value});
-    return primaryKey;
-  }
```
Comment (atomrc) [inline src/script/storage/StorageRepository.ts:79]: only consumed in tests 🙃

**Review by PatrykBuniX — APPROVED** [review]
Looks good 😄

**Pull request merged**